internal/sqlparser/parser.go

```diff
@@ -25,6 +24,37 @@ func FromBool(b bool) Direction {
 	return DirectionDown
 }
 
+type GooseError string
```

Comment:
I'm not convinced this makes things more readable.

The sql parser is an internal package, which means these errors aren't user-facing. Furthermore, we're not surfacing any specific information, i.e., it's still a `string` type which means asserting for this error still gives back the same as `.Error` would. I think this adds more indirection.